samples/Calling/src/app/utils/ShakeToSendLogs.tsx

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AzureLogger } from '@azure/logger';
+import { DefaultButton, Dialog, DialogFooter, DialogType, Link, PrimaryButton, Spinner, Text } from '@fluentui/react';
+import React from 'react';
+import { useEffect } from 'react';
+import Shake from 'shake.js';
+
+const originalConsoleLog = console.log;
+const originalAzureLoggerLog = AzureLogger.log;
+const consoleLogs: string[] = ['---- LOGS START ----'];
+
+/**
+ * Track console logs for pushing to a debug location.
+ * This is particularly useful on mobile devices where the console is not easily accessible.
+ */
+const startRecordingLogs = (): void => {
+  console.log = (...args: unknown[]) => {
```

Comment:
do these assignments mean that any console logs inside the application and lib that we put will get trapped here?

Reply:
All console logs that use console.log across the whole `window` will get caught by this function. This function does call the original console.log so they will be outputted to the console, but we just also will store them in the logs array